2.6.0 build failure with "Undefined symbols" errors

```
ar crs libr.a .libr/*.o
/Library/Developer/CommandLineTools/usr/bin/make libr.dylib
/Library/Developer/CommandLineTools/usr/bin/make libr.dylib BUILD_MERGED=1
clang -fvisibility=hidden -install_name @rpath/libr2.dylib -shared -dynamiclib -o libr.dylib  \
		.libr/*.o \
		../shlr/gdb/lib/libgdbr.a ../shlr/java/libr_java.a \
		../shlr/zip/librz.a \
		../shlr/libr_shlr.a ../shlr/ar/libr_ar.a ../shlr/grub/libgrubfs.a ../shlr/windbg/libr_windbg.a ../shlr/qnx/lib/libqnxr.a ../shlr/bochs/lib/libbochs.a\
		../shlr/capstone/libcapstone.a
Undefined symbols for architecture x86_64:
  "_BIO_ctrl", referenced from:
      _r_socket_accept in libr_socket.o
      _r_socket_flush in libr_socket.o
  "_BIO_f_buffer", referenced from:
      _r_socket_accept in libr_socket.o
  "_BIO_f_ssl", referenced from:
      _r_socket_accept in libr_socket.o
  "_BIO_new", referenced from:
      _r_socket_accept in libr_socket.o
  "_BIO_new_socket", referenced from:
      _r_socket_accept in libr_socket.o
  "_BIO_push", referenced from:
      _r_socket_accept in libr_socket.o
  "_BIO_read", referenced from:
      _r_socket_read in libr_socket.o
  "_BIO_write", referenced from:
      _r_socket_write in libr_socket.o
  "_SSL_CTX_free", referenced from:
      _r_socket_free in libr_socket.o
  "_SSL_CTX_new", referenced from:
      _r_socket_listen in libr_socket.o
  "_SSL_CTX_set_verify_depth", referenced from:
      _r_socket_listen in libr_socket.o
  "_SSL_CTX_use_PrivateKey_file", referenced from:
      _r_socket_listen in libr_socket.o
  "_SSL_CTX_use_certificate_chain_file", referenced from:
      _r_socket_listen in libr_socket.o
  "_SSL_accept", referenced from:
      _r_socket_accept in libr_socket.o
  "_SSL_free", referenced from:
      _r_socket_free in libr_socket.o
      _r_socket_close in libr_socket.o
  "_SSL_library_init", referenced from:
      _r_socket_new in libr_socket.o
  "_SSL_load_error_strings", referenced from:
      _r_socket_new in libr_socket.o
  "_SSL_new", referenced from:
      _r_socket_accept in libr_socket.o
  "_SSL_read", referenced from:
      _r_socket_read in libr_socket.o
  "_SSL_set_bio", referenced from:
      _r_socket_accept in libr_socket.o
  "_SSL_write", referenced from:
      _r_socket_write in libr_socket.o
  "_SSLv23_method", referenced from:
      _r_socket_listen in libr_socket.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [libr.dylib] Error 1
make[2]: *** [libr.dylib] Error 2
make[1]: *** [all] Error 2
make: *** [all] Error 2
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: make CS_PATCHES=0
```
Full log: https://gist.github.com/ilovezfs/ddb423ddc9aaa4f160f083c49b95e8b5

Same build failure doesn't happen with 2.5.0.

Tested on macOS 10.11, 10.12, and 10.13.

https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/24418/

R2 builds fine without ssl. In fact ssl is not used or required in r2 . Can you just drop the dependency in brew? Will evaluate if we can kill this dependency or enforce ssl builds in jwnkins to avoid this to happen

> On 22 May 2018, at 08:40, ilovezfs <notifications@github.com> wrote:
> 
> ar crs libr.a .libr/*.o
> /Library/Developer/CommandLineTools/usr/bin/make libr.dylib
> /Library/Developer/CommandLineTools/usr/bin/make libr.dylib BUILD_MERGED=1
> clang -fvisibility=hidden -install_name @rpath/libr2.dylib -shared -dynamiclib -o libr.dylib  \
> 		.libr/*.o \
> 		../shlr/gdb/lib/libgdbr.a ../shlr/java/libr_java.a \
> 		../shlr/zip/librz.a \
> 		../shlr/libr_shlr.a ../shlr/ar/libr_ar.a ../shlr/grub/libgrubfs.a ../shlr/windbg/libr_windbg.a ../shlr/qnx/lib/libqnxr.a ../shlr/bochs/lib/libbochs.a\
> 		../shlr/capstone/libcapstone.a
> Undefined symbols for architecture x86_64:
>   "_BIO_ctrl", referenced from:
>       _r_socket_accept in libr_socket.o
>       _r_socket_flush in libr_socket.o
>   "_BIO_f_buffer", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_f_ssl", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_new", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_new_socket", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_push", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_read", referenced from:
>       _r_socket_read in libr_socket.o
>   "_BIO_write", referenced from:
>       _r_socket_write in libr_socket.o
>   "_SSL_CTX_free", referenced from:
>       _r_socket_free in libr_socket.o
>   "_SSL_CTX_new", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_CTX_set_verify_depth", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_CTX_use_PrivateKey_file", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_CTX_use_certificate_chain_file", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_accept", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_SSL_free", referenced from:
>       _r_socket_free in libr_socket.o
>       _r_socket_close in libr_socket.o
>   "_SSL_library_init", referenced from:
>       _r_socket_new in libr_socket.o
>   "_SSL_load_error_strings", referenced from:
>       _r_socket_new in libr_socket.o
>   "_SSL_new", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_SSL_read", referenced from:
>       _r_socket_read in libr_socket.o
>   "_SSL_set_bio", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_SSL_write", referenced from:
>       _r_socket_write in libr_socket.o
>   "_SSLv23_method", referenced from:
>       _r_socket_listen in libr_socket.o
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make[3]: *** [libr.dylib] Error 1
> make[2]: *** [libr.dylib] Error 2
> make[1]: *** [all] Error 2
> make: *** [all] Error 2
> /usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
> BuildError: Failed executing: make CS_PATCHES=0
> Full log: https://gist.github.com/ilovezfs/ddb423ddc9aaa4f160f083c49b95e8b5
> 
> Same build failure doesn't happen with 2.5.0.
> 
> Tested on macOS 10.11, 10.12, and 10.13.
> 
> https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/24418/
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Can you submit a patch fixing this?

> On 22 May 2018, at 08:40, ilovezfs <notifications@github.com> wrote:
> 
> ar crs libr.a .libr/*.o
> /Library/Developer/CommandLineTools/usr/bin/make libr.dylib
> /Library/Developer/CommandLineTools/usr/bin/make libr.dylib BUILD_MERGED=1
> clang -fvisibility=hidden -install_name @rpath/libr2.dylib -shared -dynamiclib -o libr.dylib  \
> 		.libr/*.o \
> 		../shlr/gdb/lib/libgdbr.a ../shlr/java/libr_java.a \
> 		../shlr/zip/librz.a \
> 		../shlr/libr_shlr.a ../shlr/ar/libr_ar.a ../shlr/grub/libgrubfs.a ../shlr/windbg/libr_windbg.a ../shlr/qnx/lib/libqnxr.a ../shlr/bochs/lib/libbochs.a\
> 		../shlr/capstone/libcapstone.a
> Undefined symbols for architecture x86_64:
>   "_BIO_ctrl", referenced from:
>       _r_socket_accept in libr_socket.o
>       _r_socket_flush in libr_socket.o
>   "_BIO_f_buffer", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_f_ssl", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_new", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_new_socket", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_push", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_BIO_read", referenced from:
>       _r_socket_read in libr_socket.o
>   "_BIO_write", referenced from:
>       _r_socket_write in libr_socket.o
>   "_SSL_CTX_free", referenced from:
>       _r_socket_free in libr_socket.o
>   "_SSL_CTX_new", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_CTX_set_verify_depth", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_CTX_use_PrivateKey_file", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_CTX_use_certificate_chain_file", referenced from:
>       _r_socket_listen in libr_socket.o
>   "_SSL_accept", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_SSL_free", referenced from:
>       _r_socket_free in libr_socket.o
>       _r_socket_close in libr_socket.o
>   "_SSL_library_init", referenced from:
>       _r_socket_new in libr_socket.o
>   "_SSL_load_error_strings", referenced from:
>       _r_socket_new in libr_socket.o
>   "_SSL_new", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_SSL_read", referenced from:
>       _r_socket_read in libr_socket.o
>   "_SSL_set_bio", referenced from:
>       _r_socket_accept in libr_socket.o
>   "_SSL_write", referenced from:
>       _r_socket_write in libr_socket.o
>   "_SSLv23_method", referenced from:
>       _r_socket_listen in libr_socket.o
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make[3]: *** [libr.dylib] Error 1
> make[2]: *** [libr.dylib] Error 2
> make[1]: *** [all] Error 2
> make: *** [all] Error 2
> /usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
> BuildError: Failed executing: make CS_PATCHES=0
> Full log: https://gist.github.com/ilovezfs/ddb423ddc9aaa4f160f083c49b95e8b5
> 
> Same build failure doesn't happen with 2.5.0.
> 
> Tested on macOS 10.11, 10.12, and 10.13.
> 
> https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/24418/
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

--- a/libr/Makefile
+++ b/libr/Makefile
@@ -45,6 +45,7 @@ endif
 ifeq ($(OSTYPE),ios)
 MLFLAGS+=-install_name @rpath/libr2.dylib
 endif
+LDFLAGS+=${SSL_LDFLAGS}
 # XXX version-script only works with shlib as output
 #ifeq ($(OSTYPE),linux)
 #PLFLAGS+=--version-script ./ld.script

